src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -80,7 +80,15 @@ class Global(var currentSettings: Settings, reporter0: Reporter)
 
   import definitions.findNamedMember
   def findMemberFromRoot(fullName: Name): Symbol = rootMirror.findMemberFromRoot(fullName)
-
+  override def deferredOpenPackageModule(container: Symbol, dest: Symbol): Unit = {
+    // Some compiler runs (e.g. Toolbox and the PC) just initialise Global and then discard the Run
+    // such that the scala package object decls never get entered into the scala package
+    if ((curRun eq null) || !isGlobalInitialized || isPastPackageObjects) {
```

Comment:
I'm trying to pinpoint what breaks without the `isGlobalInitialized` condition. Ideally we can structure the initialization `Definitions` to work without the `package object scala`'s members being entered into the package.

Reply:
I'm guessing this is to fix the `type Serializable not found` errors in `ClassPrintTest`, etc.

The root cause for the problem is the toolbox compiler has an ad-hoc usage of the compiler that doesnt just run through the phases, it just calls `Typer#typed` after doing a little setup work. We should override `deferredOpenPackageModule` in `ReflectGlobal` to be eager. 

This should be test driven by a test that checks that that involves user-defined package objects. Such a test should:

  - Compile a package object into a temp dir
  - Construct a classloader with temp dir and initialize a toolbox compiler with that classloader
  - Typecheck a snippet that references a member of the package object.

Reply:
The tests that had failed with your initial patch and passed with my fixup were:

```
partest test/files/presentation/hyperlinks-macro test/files/presentation/t8934;
junit/Test/testOnly scala.lang.annotations.RunTest scala.lang.primitives.NaNTest scala.reflect.ClassOfTest scala.reflect.ClassTagTest scala.reflect.internal.QuasiTreesPrintTest;
scalacheck/Test/testOnly scala.reflect.quasiquotes.ErrorProps scala.reflect.quasiquotes.TypecheckedProps
```

So I wasn't seeing `ClassPrintTest` specifically, but yes, it was with the toolbox not running through to the package objects phase, meaning that those key modules weren't ever getting their decls copied over because a fresh run got a fresh "deferredOpen" buffer.  I wanted to find a way not to have to make things eager, because that would make it inconsistent when namer does and doesn't copy, but ended up going for that with `(curRun eq null) || !isGlobalInitialized`.  Btw, I took those as a single condition, copied from `isPast`.